cmd/push.go

```diff
@@ -26,35 +29,75 @@ var Push = func(providers app.Providers) *cli.Command {
 		},
 		Action: func(c *cli.Context) error {
 			return AppWorkspaceAction(c, func(ctx context.Context, cfg api.Config, appDetails app.Details) error {
-				source, version := c.String("source"), DetectAppVersion(c)
-				osWriters := logging.StandardOsWriters{}
-				provider := providers.FindFactory(*appDetails.Module)
-				if provider == nil {
-					return fmt.Errorf("push is not supported for this app")
+				source, version := c.String("source"), c.String("version")
+
+				pusher, err := getPusher(providers, cfg, appDetails)
+				if err != nil {
+					return err
+				}
+
+				if version == "" {
+					fmt.Fprintf(os.Stderr, "No version specified. Defaulting version based on current git commit sha...\n")
+					version, err = calcNewVersion(ctx, *pusher)
+					if err != nil {
+						return err
+					}
+					fmt.Fprintf(os.Stderr, "Version defaulted to: %s\n", version)
 				}
-				return push(ctx, cfg, appDetails, osWriters, provider, source, version)
+
+				return push(ctx, *pusher, source, version)
 			})
 		},
 	}
 }
 
-func push(ctx context.Context, cfg api.Config, appDetails app.Details, osWriters logging.OsWriters, provider *app.Provider, source, version string) error {
+func getPusher(providers app.Providers, cfg api.Config, appDetails app.Details) (*app.Pusher, error) {
+	osWriters := logging.StandardOsWriters{}
+	provider := providers.FindFactory(*appDetails.Module)
```

Comment:
You can shorten this function to:
```suggestion
	pusher, err := providers.FindPusher(osWriters, outputs.ApiRetrieverSource{Config: cfg}, appDetails)
```

Reply:
Done